torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1610,7 +1610,7 @@ void ProcessGroupNCCL::workEnqueue(
 }
 
 ProcessGroupNCCL::Options::Options(bool is_high_priority_stream)
-    : Backend::Options(NCCL_BACKEND_NAME),
+    : Backend::Options(NCCL_BACKEND_NAME, kProcessGroupNCCLDefaultTimeout),
```

Comment:
Why do we need this Options here?

Reply:
we seem to have a subclass of Options for every single entity in c10d.  idk if it's a good pattern.